client/product-details/site-messaging/index.js

```diff
@@ -0,0 +1,36 @@
+// global Stripe, wcpayStripeSiteMessaging
+function siteMessaging() {
+	const {
+		price,
+		currency,
+		country,
+		publishableKey,
+		paymentMethods,
+	} = window.wcpayStripeSiteMessaging;
+
+	// eslint-disable-next-line no-undef
+	const stripe = Stripe( publishableKey );
+	const options = {
+		amount: parseInt( price, 10 ) || 0, // In cents TODO: use quantity selector
+		currency: currency || 'USD',
+		paymentMethodTypes: paymentMethods || [],
+		countryCode: country, // Customer's country or base country of the store.
+	};
+	const elements = stripe.elements();
+	const paymentMessageElement = elements.create(
+		'paymentMethodMessaging',
+		options
+	);
+	paymentMessageElement.mount( '#payment-method-message' );
+
+	const quantitySelector = document.querySelector( '.quantity input' );
```

Comment:
What about variable products? E.g. T-shirt with size/colour selector?

Reply:
I'll take care of this in a following PR